### PR TITLE
feat: Relocate computeSystemFee and refactor getModifiedFlows

### PR DESCRIPTION
**Pull request by nicholaspai**
- `computeLpFee` shouldn't try to retrieve a bundle config from state since its expected to be used by the relayer. If the relayer is trying to set the LP fee for a very recent deposit, then it should not be using the config included at the time of the last validated bundle, instead it should be using the latest config in the `ConfigStore` contract
- `getModifiedFlows` is refactored and now is used in `resolveHistoricalDepositForFill` to reduce the DOS vector
- The only remaining DOS vector is looking up an old fill for a refund. I've left a TODO

Fixes #ACX1321

### CODE REVIEW

File: src/clients/UBAClient/UBAClientUtilities.ts
Comment (pxrl) [inline src/clients/UBAClient/UBAClientUtilities.ts:847]: What's the reason for switching to use `mapAsync` here insteead of `filterAsync` ? As far as I can see, `filterAsync` does what we want and doesn't require the subsequent loop to filter out all the `undefined` instances.

Comment (pxrl) [reply src/clients/UBAClient/UBAClientUtilities.ts:847]: Ah - I see, it's because `quoteBlockNumber`  is being added to each fill.

Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientUtilities.ts:923]: yep

File: src/clients/UBAClient/UBAClientUtilities.ts
Comment (james-a-morris) [inline src/clients/UBAClient/UBAClientUtilities.ts:315]: Is this currently just a wrapper over `resolveCorrespondingDepositForFill`

File: src/clients/UBAClient/UBAClientUtilities.ts
Comment (james-a-morris) [inline src/clients/UBAClient/UBAClientUtilities.ts:760]: Could we use `filterAsync`

**Review by james-a-morris — COMMENTED** [review]
Left comments

File: src/clients/UBAClient/UBAClientUtilities.ts
Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientUtilities.ts:315]: yeah basically

File: src/clients/UBAClient/UBAClientUtilities.ts
Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientUtilities.ts:760]: this is a `sync` filter actually. The `array` is `await getFills` and the filter function is syncrhonous

**Review by james-a-morris — APPROVED** [review]
LGTM - pending @pxrl signoff and the resolution of my outstanding comment

File: src/clients/UBAClient/UBAClientUtilities.ts
Comment (pxrl) [inline src/clients/UBAClient/UBAClientUtilities.ts]: Can we retain this filtering? It's optional to use, and the relayer relies on filtering by `relayer` to find its existing refund requests.

Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientUtilities.ts]: sure

**Pull request merged**
